### ISSUE TITLE
Wallboxen: Automatische Umschaltung der Phasenanzahl während des Ladens

### ISSUE THREAD

**Issue by camueller** [camueller/SmartApplianceEnabler#281]
Bei 1~ 11A soll eine Umschaltung in 3~ erfolgen, wenn es ein 2-phasiges Auto (VW...) ist bzw. nach 16 A auf 3~ für "echte" Drehstromlader.

![image](https://user-images.githubusercontent.com/4489628/174311281-e866d3d7-5c20-4768-b0dd-c2890768b745.png)

Übernommen aus: https://www.photovoltaikforum.com/thread/104060-ger%C3%A4te-mit-home-manager-koppeln-via-semp-ethernet/?postID=2350197#post2350197

**Comment by Trunks1982** [comment]
Das hatten wir doch auch schon besprochen.Da gibts ja eigentlich nen issue zu.

**Comment by camueller** [comment]
Welchen? (Link) Ich kenne kein Issue zu diesem Thema. Lediglicg im PV-Forum ist das Thema mal aufgekommen.

**Comment by premultiply** [comment]
Die Sache ist wesentlich komplexer als hier auf "den ersten Blick" angenommen.
Die verschiedenen Fahrzeuge haben unterschiedliche Eigenschaften von Mini- und Maximalleistungen bei unterschiedlichen Phasenkonfigurationen, erlaubte Anzahl von Umschaltungen und Ladeunterbrechungen bis Ladeverweigerung, verschiedene Phasennutzungen, zu berücksichtigende Timeouts. Dazu sind Hysteresen, Schein- vs. Wirkleistung und die damit verbundenen Nichtlinearitäten bei unterschiedlichen Phasenstromvorgaben usw. zu berücksichtigen.
Leider wirklich knifflig. Der Teufel steckt im Detail.

Ich verwende daher zur Fahrzeugladung das evcc-Projekt (https://github.com/evcc-io/evcc) parallel zum SAE der sich wie gewohnt fachgerecht um die "normalen" Hausverbraucher kümmert.
Die Fahrzeugladung wird aber bei mir ganz dem Spezialisten überlassen der u.a. das 1p3p-Feature gleich mitbringt und auch nicht auf die für die Fahrzeugladung viel zu träge interne Regelung des SHM zurückgreifen muss.
Und das schönste ist: Die ganzen Live-Fahrzeugdatenschnittstellen und diverseste Wallboxunterstützung ist gleich dabei und man muss nicht selbst mit Scripten hantieren. Und die Prognose und Verbrauchsdaten landen am Ende genau wie beim SAE im Sunny Portal.
Die Wallbox habe ich daher im SAE gar nicht konfiguriert.

Warum also nochmal neu erfinden wenn man einfach beides gleichzeitig und aus meiner Sicht optimal ergänzend haben kann?
Oder den Code abschreiben 😅

**Comment by camueller** [comment]
Einige Links zum Thema:
https://www.e-mobileo.de/warum-automatische-phasenumschaltung/
https://openwb.de/forum/viewtopic.php?f=8&t=1727

Basierend auf diesen Informationen wird der SAE ermöglichen
- eine Aktion zum Aktivieren des 3-phasigen Laden zu definieren
- eine Aktion zum Aktivieren des 1-phasigen Laden zu definieren
- eine Leistungsschwelle zu definieren, bei deren Überschreiten das 3-phasige Laden aktiviert wird und bei deren Unterschreiten das 1-phasige Laden aktiviert wird

Alles andere muss die Wallbox selbst regeln.

**Comment by RaphiBu** [comment]
Gibt es eine Wallbox, die das kann? Die Warp 2 z.B. kann das selbst nicht. Tinkerforge entwickelt momentan einen Lastmanager mit einem Relais, mit dem man dann im Verteiler 2 Phasen abschalten kann. Mit Deiner Lösung müsste ich wahrscheinlich versuchen, einen Ausgang des Raspberry PI zu nutzen, um ein Relais zu schalten, das mir dann über ein Schütz zwei Phasen schaltet, oder?

**Comment by premultiply** [comment]
Ja, siehe https://docs.evcc.io/docs/devices/chargers
Ist jeweils bei den einzelnen Boxen mit "Die 1P/3P-Phasenumschaltung wird unterstützt" vermerkt.

Die Variante mit einem externen Lasttrennschalter bei allen anderen Boxen ist allerdings auch durchaus üblich.

**Comment by Trunks1982** [comment]
Ja zb go e charger v3

**Comment by Trunks1982** [comment]
> Welchen? (Link) Ich kenne kein Issue zu diesem Thema. Lediglicg im PV-Forum ist das Thema mal aufgekommen.

Das hatten wir beide schon einmal besprochen gehabt das du es integrieren wolltest.

**Comment by RaphiBu** [comment]
> Die Variante mit einem externen Lasttrennschalter bei allen anderen Boxen ist allerdings auch durchaus üblich.

Könnte ich meine Warp 2 Pro damit aufrüsten?

![grafik](https://user-images.githubusercontent.com/107432815/174455094-b2d3d189-7b8a-4400-ad5c-43c5e213be9d.png)

Bzw. mit etwas ähnlichem, da der hier nur insgesamt 25A schalten mag, ich aber bei zwei Phasen 32A bräuchte :-(

**Comment by premultiply** [comment]
Hager HAB304

**Comment by camueller** [comment]
> Hager HAB304

Der ist aber nicht steuerbar (z.B. via HTTP).

**Comment by premultiply** [comment]
Absichtlich nicht.
Elektronisch ansteuerbare Schalter würden sonst im Fehlerfall (Software, Kommunikation oder Hardware) bei manchen Fahrzeugen das Onboard-Ladesystem zerstören. Das ergibt dann ggf. mehrere tausend Euro Schaden.
Ich denke nicht dass du die gerne übernehmen möchtest.
Ich kann nur raten: Finger weg von solchen Experimenten.

Wenn dann nur über Wallboxen die den Umschaltvorgang intern definiert und sicher selbst abwickeln.

**Comment by camueller** [comment]
Dann kann er den Schalter auch weglassen und immer einphasig laden. Das ist für die meisten PV-Anlagen (habe ja meist < 10 kWp) eh das Beste.

**Comment by premultiply** [comment]
So ist es.
Sollte man aber doch mal schneller mit Netzstrom laden wollen ist so ein günstiger manueller Schalter eine sehr gute Alternative.

**Comment by RaphiBu** [comment]
> Absichtlich nicht.
Elektronisch ansteuerbare Schalter würden sonst im Fehlerfall (Software, Kommunikation oder Hardware) bei manchen Fahrzeugen das Onboard-Ladesystem zerstören. Das ergibt dann ggf. mehrere tausend Euro Schaden.
Ich denke nicht dass du die gerne übernehmen möchtest.
Ich kann nur raten: Finger weg von solchen Experimenten. Wenn dann nur über Wallboxen die den Umschaltvorgang intern definiert und sicher selbst abwickeln.

Da hast Du recht, danke für den Hinweis :-)

> Dann kann er den Schalter auch weglassen und immer einphasig laden. Das ist für die meisten PV-Anlagen (habe ja meist < 10 kWp) eh das Beste.

In der Wallbox und im Verteiler sind jeweils Sicherungsautomaten, für jede Phase einzeln. Dann kann ich doch einfach L2 und L3 damit abschalten, oder? Für das "Schnellladen" habe ich noch eine zweite, ungeregelte Wallbox im Carport, für den Notfall ;-)

**Comment by premultiply** [comment]
Ja, kann man so machen - auch wenn das elektrotechnisch so nicht sein sollte.

Eigentlich muss hier sichergestellt sein dass alle 3 Phasen in einer Überlastsituation gleichzeitig abgeschaltet werden. Da gehört daher eigentlich ein 3-phasiger Automat hin und wie skiziert ein Lasttrennschalter für L2+L3 (zusammen) dahinter. Dann ist sichergestellt dass es nur die Modi 1P und 3P gibt. Andere Kombinationen können nämlich auch zu Problem führen...

**Comment by RaphiBu** [comment]
In der Wallbox ist ein 3-phasiger Automat verbaut, im Verteiler jedoch nicht.

**Comment by e-Surfer** [comment]
> Ja, kann man so machen - auch wenn das elektrotechnisch so nicht sein sollte.
> 
> 
> 
> Eigentlich muss hier sichergestellt sein dass alle 3 Phasen in einer Überlastsituation gleichzeitig abgeschaltet werden. Da gehört daher eigentlich ein 3-phasiger Automat hin und wie skiziert ein Lasttrennschalter für L2+L3 (zusammen) dahinter. Dann ist sichergestellt dass es nur die Modi 1P und 3P gibt. Andere Kombinationen können nämlich auch zu Problem führen...

Hallo zusammen,

das heißt mit L1 nach dem Sicherungsautomat direkt zur Wallbox. L2 und L3 ebenfalls nach dem Sicherungsautomat, dann jedoch über den Lasttrennschalter, zur Wallbox. Beim Umschalten des Lasttrennschalter werden L2 und L3 zeitgleich getrennt und L1 liefert weiterhin dauerhaft ohne Unterbrechung Strom zur Wallbox. Somit läuft die Wallbox unterbrechungsfrei weiter.

Vor Umschaltung sicherstellen, dass Auto nicht angeschlossen.

Habe ich das so korrekt verstanden?

Viele Grüße
Christian

**Comment by camueller** [comment]
Hier ist ein sehr umfrangreicher Beitrag inkl. der notwendigen Hardware:
https://www.meintechblog.de/2021/12/pimp-my-wallbox-phasenumschaltung-nachruesten-aber-wie/

**Comment by camueller** [comment]
Siehe auch: https://forum.iobroker.net/post/947726

**Comment by ArminCoin** [comment]
Gibt es hierzu evtl. schon einen neuen Status, der go-e Charger 2, kann die Phasenumschaltung beispielsweise ja schon per API?

**Comment by camueller** [comment]
Nein, hat auch keine hohe Priorität, da komplexes Thema und Überschussladen einphasig meist sowieso am sinnvollsten ist.

**Comment by ArminCoin** [comment]
> Nein, hat auch keine hohe Priorität, da komplexes Thema und Überschussladen einphasig meist sowieso am sinnvollsten ist.

Okay, ist denn evtl. das manuelle Umschalten geplant? Aktuell lade ich per Überschuss immer aus 1 Phase und zum schnellen Vollladen auf 3 Phasen. Es wäre super wenn man das irgendwie im SAE einstellen könnte. Aktuell habe ich damit etwas Probleme wenn ich z.B. vergesse manuell wieder auf 1 Phase zu schalten klappt das Überschussladen nicht richtig.

**Comment by camueller** [comment]
> Okay, ist denn evtl. das manuelle Umschalten geplant?

Nein.

**Comment by RaphiBu** [comment]
Hallo Axel,

 

vielen Dank für Deine Nachricht. War in letzter Zeit ziemlich stressig, sorry für die späte Antwort.

Ich habe mich inzwischen mit der einen Phase abgefunden. Maximal kommen sowieso nur ca. 5 kW vom Dach, das passt also im Grunde auch.

 

Inzwischen funktioniert alles sehr robust und ich erfreue mich eines hohen EV-Anteils (ca. 36% letztes Jahr).

 

Danke Dir für Deine Unterstützung :-)

 

Viele Grüße,

 

Raphael

 
 

Gesendet: Samstag, 11. Februar 2023 um 21:10 Uhr
Von: "Axel Müller" ***@***.***>
An: "camueller/SmartApplianceEnabler" ***@***.***>
Cc: "RaphiBu" ***@***.***>, "Comment" ***@***.***>
Betreff: Re: [camueller/SmartApplianceEnabler] Wallboxen: Automatische Umschaltung der Phasenanzahl während des Ladens (Issue #281)


 

Hier ist ein sehr umfrangreicher Beitrag inkl. der notwendigen Hardware:
https://www.meintechblog.de/2021/12/pimp-my-wallbox-phasenumschaltung-nachruesten-aber-wie/

**Comment by camueller** [comment]
Da ich seit einigen Monaten selbst ein E-Auto mit 3-phasigem Lader besitze, habe ich mich auch mit diesem Thema befast. 

Der Sinn des "Umschaltens" der Phasenanzahl besteht ja darin, dass das Überschussladen nicht erst bei 3*6A=18A bzw. 4140W beginnen soll (6A ist die minimale Stromstäke je Phase). Andererseits soll das Sofortladen mit maximaler Stromstärke möglich sein.

Folgende Lösung nutze ich:
Ich habe im SAE zwei Konfigurationen für mein Fahrzeug angelegt (in dieser Reihenfolge, das ist wichtig, weil die erste Konfiguration automatisch beim Verbinden gewählt wird).

1. die Konfiguraiton für Überschussladen, d.h. reduzierte Phasenanzahl. In meinem Fall sind das 2 Phasen. Dementsprechend ist die minimale Leistungsaufnahme in der Gerätekonfiguration im SAE für die Wallbox auf 2*6A=12A bzw. 2760W gesetzt, d.h. der Home Manger schickt nur Einschalt-Befehle mit mindestens dieser Leistung. Bei einem großen Akku sind geringere Ladeleistungen aus meiner Sicht sowieso nicht wirklich hilfreich. Natürlich muss dafür bei einer Phase manuell die Sicherung im Schaltschrank ausgeschaltet werden!
2. Die Konfiguration für Sofortladen, d.h. 3 Phasen. Diese Konfiguration muss ja sowieso manuell gewählt werden. Laden kann über Sofortladen oder zeitgesteuertes Laden (zur Nutzung dynamischer Stromtarife nachts) gestartet werden. Die konfigurierte, minimale Leistungsaufnahme spielt keine Rolle, da es um das Laden mit maximaler Stromstärke geht. Natürlich muss dafür auch bei der dritten 3 Phase manuell die Sicherung im Schaltschrank eingeschaltet werden!

Das manuelle Ein-/Ausschalten einer Sicherung im Schaltschrank finde ich für mich persönlich nicht schlimm, da ich über das Jahr gesehen > 80% Überschußladen mache und nur in Ausnahmefällen Strom aus dem Netz 3-phasig beziehen möchte. Übrigens hindert einen niemand daran, auch beim Laden mit Netzstrom 2-phasig zu laden, dann entfällt das beschriebene Umschalten der Sicherung und man benötigt auch nicht die zwei Konfigurationen im SAE.

Zu beachten ist, dass verschiedene 3-phasige Zähler je nach Modell mindestens 1 oder sogar 2 Phasen benötigen, um korrekt zu funktionieren. Als ich mit nur einer Phase getestet hatte und dabei feststellte, dass der Zähler nicht funktioniert, habe ich in dessen Anleitung nachgesehen. Daraus war ersichtlich, dass er mindestens 2 Phasen braucht. Daraufhin hatte ich mir einen Zähler bestellt, der mit 1 oder 3 Phasen arbeiten kann, aber diesen wieder zurückgesendet, nachdem mit klar wurde, dass ich nicht einphasig Überschußladen machen möchte, weil ich dann bei der maximalen Ladeleistung zu stark limitiert bin.

Der SAE wird auch künftig keinen Mechanismus zur automatischen Umschaltung der Phasenanzahl haben.
